x/ecocredit/client/tx.go

```diff
@@ -504,3 +509,224 @@ Parameters:
+
+			// loop through orders with ask price as string
+			for i, order := range strOrders {
+
+				// parse and normalize ask price as sdk.Coin
+				askPrice, err := sdk.ParseCoinNormalized(order.AskPrice)
+				if err != nil {
+					return err
+				}
+
+				// set order with ask price as sdk.Coin
+				orders[i] = &ecocredit.MsgSell_Order{
+					BatchDenom:        order.BatchDenom,
+					AskPrice:          &askPrice,
+					Quantity:          order.Quantity,
+					DisableAutoRetire: order.DisableAutoRetire,
+				}
+			}
+
+			// create sell message
+			msg := ecocredit.MsgSell{
+				Owner:  owner.String(),
+				Orders: orders,
+			}
+
+			// generate and broadcast transaction
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	})
+}
+
+// TxUpdateSellOrdersCmd returns a transaction command that creates sell orders.
+func TxUpdateSellOrdersCmd() *cobra.Command {
+	return txflags(&cobra.Command{
+		Use:   "update-sell-orders [updates]",
+		Short: "Updates existing sell orders with transaction author (--from) as owner",
+		Long: fmt.Sprintf(
+			`Updates existing sell orders with transaction author (--from) as owner.
+
+Parameters:
+  updates:  YAML encoded update list. Note: numerical values must be written in strings.
+           eg: '[{sell_order_id: "1", new_quantity: "5", new_ask_price: "200regen", disable_auto_retire: false}]'`,
+		),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := sdkclient.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			// get the order owner from the --from flag
+			owner := clientCtx.GetFromAddress()
+
+			// declare updates array with ask price as string
+			var strUpdates []struct {
+				SellOrderId       string `json:"sell_order_id"`
+				NewQuantity       string `json:"new_quantity"`
+				NewAskPrice       string `json:"new_ask_price"`
+				DisableAutoRetire bool   `json:"disable_auto_retire"`
+			}
+
+			// unmarshal YAML encoded updates with new ask price as string
+			if err := yaml.Unmarshal([]byte(args[0]), &strUpdates); err != nil {
+				return err
+			}
+
+			// declare updates array with new ask price as sdk.Coin
+			updates := make([]*ecocredit.MsgUpdateSellOrders_Update, len(strUpdates))
+
+			// loop through updates with new ask price as string
+			for i, update := range strUpdates {
+
+				// parse sell order id
+				sellOrderId, err := strconv.ParseUint(update.SellOrderId, 10, 64)
+				if err != nil {
+					return ecocredit.ErrInvalidSellOrder.Wrap(err.Error())
+				}
+
+				// parse and normalize new ask price as sdk.Coin
+				askPrice, err := sdk.ParseCoinNormalized(update.NewAskPrice)
+				if err != nil {
+					return err
+				}
+
+				// set update with new ask price as sdk.Coin
+				updates[i] = &ecocredit.MsgUpdateSellOrders_Update{
+					SellOrderId:       sellOrderId,
+					NewAskPrice:       &askPrice,
+					NewQuantity:       update.NewQuantity,
```

Comment:
we should check this value

Reply:
We already do in the message implementation. I don't think we need to duplicate the verification here. See other tx commands for example and test cases within this pull request.